### ISSUE TITLE
Metric with multiple input runs in an unexpected way.

### ISSUE THREAD

**Issue by lyhyl** [pytorch/ignite#2940]
## ❓ Questions/Help/Support

My customized loss requires two pairs of input:
```python
class MyLoss(nn.Module):
    def __init__(self, ca: float = 1.0, cb: float = 1.0) -> None:
        super().__init__()
        self.ca = ca
        self.cb = cb

    def forward(self, y_pred: Tuple[torch.Tensor, torch.Tensor], y_true: Tuple[torch.Tensor, torch.Tensor]) -> torch.Tensor:
        a_true, b_true = y_true
        a_pred, b_pred = y_pred
        return self.ca * F.mse_loss(a_pred, a_true) + self.cb * F.cross_entropy(b_pred, b_true)

```

When I try to log the loss with `Loss` metric:
```python
loss = MyLoss(0.5, 1.0)
metrics = {
    "Loss": Loss(loss)
}
train_evaluator = create_supervised_evaluator(model, metrics, device, prepare_batch=prepare_batch)
```

It will crash on line:
https://github.com/pytorch/ignite/blob/4825bb6ecd5f432f0423bf3d587fbce04b9d2d4f/ignite/metrics/metric.py#L308
because it treats all inputs as independent pair of  y_pred and y, which is not what `MyLoss` need.

I dug into the source code I found https://github.com/pytorch/ignite/pull/2055 introduces a new feature, which causes this issue.
So, what are the best practices for dealing with multiple input losses?
